clang/lib/3C/DeclRewriter.cpp

```diff
@@ -88,13 +148,38 @@ void DeclRewriter::rewriteDecls(ASTContext &Context, ProgramInfo &Info,
         if (VDLToStmtMap.find(D) != VDLToStmtMap.end())
           DS = VDLToStmtMap[D];
 
+        // Decided to create a global variable inside PVConstraint temporarily
+        // to forego changing mkString's signature.
+        PV->RewritingBaseType = inlinestructdefs[D];
         std::string newTy = getStorageQualifierString(D) +
                             PV->mkString(Info.getConstraints().getVariables()) +
                             ABRewriter.getBoundsString(PV, D);
-        if (auto *VD = dyn_cast<VarDecl>(D))
-          RewriteThese.insert(new VarDeclReplacement(VD, DS, newTy));
-        else if (auto *FD = dyn_cast<FieldDecl>(D))
+        if (auto *VD = dyn_cast<VarDecl>(D)) {
+            RewriteThese.insert(new VarDeclReplacement(VD, DS, newTy));
+        }
+        else if (auto *FD = dyn_cast<FieldDecl>(D)) {
+          // For an inline struct, line 162 does literally nothing
           RewriteThese.insert(new FieldDeclReplacement(FD, DS, newTy));
+          auto inlinedecl = inlines.begin();
+          // Here's the idea: If a FieldDecl coincides with some inline struct
+          // VarDecl, let's replace what we WOULD rewrite the FieldDecl for
+          // in the actual struct definition associated with the VarDecl
+          while (inlinedecl != inlines.end()) {
+            if (FD->getEndLoc().getRawEncoding() <=
+                    (*inlinedecl)->getEndLoc().getRawEncoding() &&
+                FD->getBeginLoc().getRawEncoding() >=
+                    (*inlinedecl)->getBeginLoc().getRawEncoding()) {
+              // I currently use a helper function tucked away in Utils.cpp
+              // but I think using regex_replace might be more robust, but I couldn't
+              // get it to work just yet
+              // inlinestructdefs[*inlinedecl] = std::regex_replace(inlinestructdefs[*inlinedecl], std::regex(getSourceText(D->getSourceRange(), Context)), newTy);
```

Comment:
The use of regexes seems dicey to me. I don't understand why you don't know precisely what you'd be replacing, i.e., the exact name. If you are searching through a random string for something like `struct` then that's not robust; there could be code in comments, variable names, etc. that accidentally matches.